src/backend/brilisp.py

```diff
@@ -26,7 +26,10 @@ def gen_function(expr):
 
 def gen_type(typ):
     if is_list(typ):
-        return {typ[0]: gen_type(typ[1])}
+        retval = {typ[0]: gen_type(typ[1])}
+        if len(typ) > 2 and is_list(typ[2]):
```

Comment:
assert that `typ[0]` is ptr is it's a list. also check that typ[2][1] is addrspace.

Reply:
Done